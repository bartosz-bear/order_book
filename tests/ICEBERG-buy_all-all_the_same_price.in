{"type": "Iceberg", "order": {"direction": "Buy", "id": 1, "price": 14, "quantity": 20, "peak": 3}}
{"type": "Iceberg", "order": {"direction": "Buy", "id": 2, "price": 14, "quantity": 20, "peak": 3}}
{"type": "Iceberg", "order": {"direction": "Buy", "id": 3, "price": 14, "quantity": 20, "peak": 3}}
{"type": "Iceberg", "order": {"direction": "Sell", "id": 4, "price": 14, "quantity": 80, "peak": 4}}