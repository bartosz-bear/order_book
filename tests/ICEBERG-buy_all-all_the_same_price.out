{"sellOrders": [], "buyOrders": [{"id": 1, "price": 14, "quantity": 3}]}
{"sellOrders": [], "buyOrders": [{"id": 1, "price": 14, "quantity": 3}, {"id": 2, "price": 14, "quantity": 3}]}
{"sellOrders": [], "buyOrders": [{"id": 1, "price": 14, "quantity": 3}, {"id": 2, "price": 14, "quantity": 3}, {"id": 3, "price": 14, "quantity": 3}]}
{"sellOrders": [{"id": 4, "price": 14, "quantity": 4}], "buyOrders": []}
{"buyOrderId": 1, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 2, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 3, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 1, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 2, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 3, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 1, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 2, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 3, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 1, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 2, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 3, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 1, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 2, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 3, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 1, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 2, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 3, "sellOrderId": 4, "quantity": 3, "price": 14}
{"buyOrderId": 1, "sellOrderId": 4, "quantity": 2, "price": 14}
{"buyOrderId": 2, "sellOrderId": 4, "quantity": 2, "price": 14}
{"buyOrderId": 3, "sellOrderId": 4, "quantity": 2, "price": 14}
## jeżeli pojawi się quantity=1, to znaczy, że agresywny jest emulowany przez pasywny; tj. peak jest uwzględniany w agresywnym orderze - a nie powinien być